### ISSUE TITLE
Change way of running negative tests

### ISSUE THREAD

**Issue by gireeshpunathil** [appsody/appsody#331]
**Is your feature request related to a problem? Please describe.**

Right now navigating through the test results are bit troublesome, owing to the misleading output that come from negative tests. For example:

```
=== RUN   TestRepoAddErrors/One_arg
[go run /home/travis/gopath/src/github.com/appsody/appsody/cmd/.. -v repo add reponame]
[Debug] Logging to file /home/travis/.appsody/logs/appsody2019-09-17T04-25-49.406048014.log
[Error] Error, you must specify repository name and URL
Error, you must specify repository name and URL

github.com/appsody/appsody/cmd.glob..func15
	/home/travis/gopath/src/github.com/appsody/appsody/cmd/repo_add.go:34
github.com/appsody/appsody/vendor/github.com/spf13/cobra.(*Command).execute
	/home/travis/gopath/src/github.com/appsody/appsody/vendor/github.com/spf13/cobra/command.go:762
github.com/appsody/appsody/vendor/github.com/spf13/cobra.(*Command).ExecuteC
	/home/travis/gopath/src/github.com/appsody/appsody/vendor/github.com/spf13/cobra/command.go:852
github.com/appsody/appsody/vendor/github.com/spf13/cobra.(*Command).Execute
	/home/travis/gopath/src/github.com/appsody/appsody/vendor/github.com/spf13/cobra/command.go:800
github.com/appsody/appsody/cmd.Execute
	/home/travis/gopath/src/github.com/appsody/appsody/cmd/root.go:160
main.main
	/home/travis/gopath/src/github.com/appsody/appsody/main.go:26
runtime.main
	/home/travis/.gimme/versions/go1.12.9.linux.amd64/src/runtime/proc.go:200
runtime.goexit
	/home/travis/.gimme/versions/go1.12.9.linux.amd64/src/runtime/asm_amd64.s:1337
exit status 1
```

**Describe the solution you'd like**
This is happening because `cmd/cmdtest/testutils.go:RunAppsodyCmdExec` prints the standard stream data to its console unconditionally.

It would be great to control this, and suppress errors when negative tests are run.

**Describe alternatives you've considered**
1. Supress stdout and stderr always in `RunAppsodyCmdExec`, let the caller print if required.
2. Define a flag to the API - to decide print or not to print.

**Additional context**
none

**Comment by gireeshpunathil** [comment]
happy to work on it myself, but would like to see views on the 2 proposals!

**Comment by chilanti** [comment]
I find the output of build (and test) really hard to figure out too, especially in travis. All tests are run in verbose mode - which increases the confusion. And, yes, negative tests (those that are meant to produce an error) do look like they are failing. Ideally, we would want to print out the messages only if a test is failing, and that's probably easier said than done. Wondering if there's a way to "cache" the stdout/stderr and just flush them out if the test actually failed...?

**Comment by kylegc** [comment]
Ultimately RunAppsodyCmd() and all the test should be using `t.log()` to print messages. This would allow us to run parallel tests in the future and capture the output correctly.

For the time being, issue #493 is very similar to this and accomplishes this by manipulating console output.

**Comment by neeraj-laad** [comment]
@upLukeWeston @tnixa is this still an issue? I know we changed the way tests are logging output now but wanted to double-check before closing.

**Comment by upLukeWeston** [comment]
This was resolved by #657.
